proposals/collection-expressions-inline.md

```diff
@@ -0,0 +1,124 @@
+# Collection expressions: inline collections
+
+## Summary
+
+Support collection expressions *inline* in expression contexts where the *collection type* is not observable.
+
+## Motivation
+
+Inline collection expressions could be used with spreads to allow adding elements *conditionally* to the containing collection:
+```csharp
+int[] items = [x, y, .. b ? [z] : []];
+```
+
+Inline collection expressions could be used directly in `foreach`:
+```csharp
+foreach (var b in [false, true]) { }
+```
+In these cases, the *collection type* of the inline collection expression is unspecified. The choice of how or whether to instantiate the collection is left to the compiler.
+
+## Detailed design
+
+### Conversions
+
+To support conversions for inline collection expressions, an abstract *collection_type* is introduced.
```

Comment:
`abstract` is a bit ambiguous.  I would say `anonymous`.

Reply:
Replaced "abstract" with "compile time".